ldclient/impl/datasystem/fdv1.py

```diff
@@ -0,0 +1,171 @@
+        self._config = config
+
+        # Set up data store plumbing
+        self._data_store_listeners = Listeners()
+        self._data_store_update_sink = DataStoreUpdateSinkImpl(
+            self._data_store_listeners
+        )
+        # Import here to avoid circular import
+        from ldclient.client import _FeatureStoreClientWrapper
+
+        self._store_wrapper: FeatureStore = _FeatureStoreClientWrapper(
+            self._config.feature_store, self._data_store_update_sink
+        )
+        self._data_store_status_provider_impl = DataStoreStatusProviderImpl(
+            self._store_wrapper, self._data_store_update_sink
+        )
+
+        # Set up data source plumbing
+        self._data_source_listeners = Listeners()
+        self._flag_change_listeners = Listeners()
+        self._flag_tracker_impl = FlagTrackerImpl(
+            self._flag_change_listeners,
+            lambda key, context: None,  # Replaced by client to use its evaluation method
+        )
+        self._data_source_update_sink = DataSourceUpdateSinkImpl(
+            self._store_wrapper,
+            self._data_source_listeners,
+            self._flag_change_listeners,
+        )
+        self._data_source_status_provider_impl = DataSourceStatusProviderImpl(
+            self._data_source_listeners, self._data_source_update_sink
+        )
+
+        # Ensure v1 processors can find the sink via config for status updates
+        self._config._data_source_update_sink = self._data_source_update_sink
+
+        # Update processor created in start(), because it needs the ready Event
+        self._update_processor: Optional[UpdateProcessor] = None
+
+        # Diagnostic accumulator provided by client for streaming metrics
+        self._diagnostic_accumulator = None
+
+        # Track current data availability
+        self._data_availability: DataAvailability = (
+            DataAvailability.CACHED
+            if getattr(self._store_wrapper, "initialized", False)
+            else DataAvailability.DEFAULTS
+        )
+
+        # React to data source status updates to adjust availability
+        def _on_status_change(status: DataSourceStatus):
+            if status.state == DataSourceState.VALID:
+                self._data_availability = DataAvailability.REFRESHED
+
+        self._data_source_status_provider_impl.add_listener(_on_status_change)
+
+    def start(self, set_on_ready: Event):
+        """
+        Starts the v1 update processor and returns immediately. The provided
+        Event is set by the processor upon first successful initialization or
+        upon permanent failure.
+        """
+        update_processor = self._make_update_processor(
+            self._config, self._store_wrapper, set_on_ready
+        )
+        self._update_processor = update_processor
+        update_processor.start()
+
+    def stop(self):
+        if self._update_processor is not None:
+            self._update_processor.stop()
```

Comment:
I don't fully know the logic here, but since we are setting the `_update_processor` in the start method should we set it back to None after it stops? Would there ever be a case where you could call the start method on a previously stopped processor? It probably doesn't matter in this code since the start method will just overwrite it.

Reply:
I don't think it will matter. As you point out, calling start again would override it. But no one is going to be interacting with the datasystem directly. That's all internal to the SDK so shouldn't be an issue.